File: jsimple-json/c#/jsimple/json/text/JsonParsingException.cs
namespace jsimple.json.text
{

	/// <summary>
	/// @author Bret Johnson
	/// @since 5/6/12 2:13 AM
	/// </summary>
	public sealed class JsonParsingException : JsonException
	{
		public JsonParsingException(string message) : base(message)
		{
		}

		public JsonParsingException(string expected, string encountered) : base("Expected {} but encountered {}", expected, encountered)
		{
		}

		public JsonParsingException(string expected, Token token) : this(expected, token.Description)
		{
		}
	}

}

File: jsimple-json/src/main/java/jsimple/json/text/JsonParsingException.java
/*
 * Copyright (c) 2012-2014, Microsoft Mobile
 *
 * Permission is hereby granted, free of charge, to any person obtaining a copy
 * of this software and associated documentation files (the "Software"), to deal
 * in the Software without restriction, including without limitation the rights
 * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
 * copies of the Software, and to permit persons to whom the Software is
 * furnished to do so, subject to the following conditions:
 *
 * The above copyright notice and this permission notice shall be included in
 * all copies or substantial portions of the Software.
 *
 * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
 * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
 * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
 * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
 * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
 * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
 * THE SOFTWARE.
 */

package jsimple.json.text;

import jsimple.json.JsonException;

/**
 * @author Bret Johnson
 * @since 5/6/12 2:13 AM
 */
public final class JsonParsingException extends JsonException {
    public JsonParsingException(String message) {
        super(message);
    }

    public JsonParsingException(String expected, String encountered) {
        super("Expected {} but encountered {}", expected, encountered);
    }

    public JsonParsingException(String expected, Token token) {
        this(expected, token.getDescription());
    }
}
